.gitlab/ci/global.yml

```
@@ -68,3 +68,156 @@
   extends:
     - .default-cache
     - .default-before_script
+
+.notify-slack:
+  after_script:
+    - echo "=====Notify Slack====="
+    - |
+      if [ -n "${SLACK_ONLY_IF}" ]; then
+        ./Tests/scripts/slack_notifier.sh "$SLACK_TEST_TYPE" "$SLACK_ENV_RESULTS" "$SLACK_CHANNEL"
+      else
+        echo "Skipping Slack"
+      fi
+
+run-unittests-and-lint:
```

```suggestion
.run-unittests-and-lint:
```
Let's make this a hidden job such that the different workflows will create a job that will extend from this one whenever they need to run unit tests and lint.